### PR TITLE
Redirect to default source when deleting sources

### PR DESCRIPTION
**Pull request by timraymond**
- [x] CHANGELOG.md updated
  - [X] Rebased/mergable
  - [x] Tests pass
  - [X] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

Connect #1008 

### The problem

Users that deleted a source they were connected to would be unexpectedly redirected to the create sources page.

### The Solution

Users are redirected to the default source when deleting sources. If the default source is deleted, the API chooses a new default source and the user is directed to that one. If all sources are deleted, the user is prompted to create a new one.

### CODE REVIEW

File: ui/src/shared/actions/sources.js [hunk 1]
```diff
@@ -1,35 +1,30 @@
...
-export function addSource(source) {
-  return {
-    type: 'SOURCE_ADDED',
-    payload: {
-      source,
-    },
-  };
+export const removeAndLoadSources = (source) => async (dispatch) => {
+  await deleteSource(source)
+  const sources = await getSources()
```
Comment (jaredscheib) [inline ui/src/shared/actions/sources.js]: syntactic convention here has been to peel off whatever data, ex. 'const {data}' or even 'const {data: {sources}}'

File: ui/src/shared/actions/sources.js [hunk 2]
```diff
@@ -1,35 +1,30 @@
...
 
-export function addSource(source) {
-  return {
-    type: 'SOURCE_ADDED',
-    payload: {
-      source,
-    },
-  };
+export const removeAndLoadSources = (source) => async (dispatch) => {
+  await deleteSource(source)
```
Comment (jaredscheib) [inline ui/src/shared/actions/sources.js]: nice use of async/await. would be good to use a try/catch as well so not only handling success case

**Comment by lukevmorris** [comment]
Ah, two tiny things:
1. I think `sources` isn't used anymore in `removeAndLoadSources`
2. The comment still mentions the removed precheck

**Comment by lukevmorris** [comment]
@timraymond thanks for the updates. Merging, since all feedback has been addressed.

**Pull request merged**